VoodooGPIO/VoodooGPIOAMD.cpp

```diff
@@ -436,20 +466,20 @@ void VoodooGPIOAMD::stop(IOService *provider) {
     IOLog("%s::VoodooGPIO stop!\n", this->getName());
 
     if (this->command_gate) {
-        this->workLoop->removeEventSource(this->command_gate);
-        OSSafeReleaseNULL(this->command_gate);
+        if (this->workLoop) {
+            this->workLoop->removeEventSource(this->command_gate);
+        }
+        this->command_gate->release();
     }
 
-    if (this->mmap) {
-        OSSafeReleaseNULL(mmap);
-    }
+    OSSafeReleaseNULL(mmap);
 
     if (this->registered_pin_list) {
         if (this->registered_pin_list->getCount() > 0) {
             IOLog("%s::Interrupt has not been unregistered by client\n", this->getName());
             this->getProvider()->unregisterInterrupt(0);
         }
-        OSSafeReleaseNULL(this->registered_pin_list);
+        this->registered_pin_list->release();
```

Comment:
`OSSafeReleaseNULL` should be kept as the macro will release and set the pointer to NULL.

Reply:
Ah, nice catch!